run_linux.go

```diff
@@ -680,6 +680,11 @@ func runUsingRuntime(isolation Isolation, options RunOptions, configureNetwork b
 		runtime = util.Runtime()
 	}
 
+	localRuntime := util.FindLocalRuntime(runtime)
+	if localRuntime != "" {
+		runtime = localRuntime
+	}
+
```

Comment:
Whoa, pretty sure we don't want to override a value that was passed to our API with a default from the configuration file.

Reply:
What's coming back from `util.Runtime()` is either `runc` or `crun`.  In some systems, most likely with an ill-defined `$PATH` variable, they are failing as they can't find the runtime to run it.

`FindLocalRuntime` retrieves a map of the runtime passed in ("runc" or "crun") from containers.conf which has the possible locations of the runtime on the system.  It then searches the locations until it finds the first one off the list.

This is similar to what libpod does, but they tuck the retrieved value away differently than we do and also get it a bit more frequently than we do.

Happy to adjust as necessary, I was just trying to model Libpod/Podman's functionality here for this.

Reply:
I'm talking about the `options.Runtime` field (line 678).  If that's set, we need to honor it.  The current version of this patch overwrites it unconditionally.

Reply:
Ah, missed that point.  Good catch.  I'll add a check to make sure `options.Runtime` is empty before doing this processing.  TYVM!